test/integration/provider/rti.go

```diff
@@ -0,0 +1,149 @@
+package provider
+
+import (
+	"fmt"
+
+	provider "github.com/gardener/machine-controller-manager-provider-azure/pkg/azure"
+	v1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+)
+
+var (
+	// cluster tag and cluster values are specifically used for integration test
+	// in case the test is run against non seed cluster then the supplied MachineClass
+	// is expected to have these tags set so that the machines from this suite won't be
+	// orphan collected.
+	clusterTag      = "mcm-integration-test"
+	clusterTagValue = "True"
+)
+
+// ResourcesTrackerImpl implements the Resource Tracker Interface from the Integration test suite
+type ResourcesTrackerImpl struct {
+	ClusterName      string
+	InitialVolumes   []string
+	InitialInstances []string
+	InitialMachines  []string
+	MachineClass     *v1alpha1.MachineClass
+	ResourceGroup    string
+	SecretData       map[string][]byte
+}
+
+// InitializeResourceTracker is the constructor of ResourceTrackerImpl
+func (r *ResourcesTrackerImpl) InitializeResourcesTracker(machineClass *v1alpha1.MachineClass, secretData map[string][]byte, clusterName string) error {
+
+	r.MachineClass = machineClass
+	r.SecretData = secretData
+	r.ClusterName = clusterName
+
+	providerSpec, err := provider.DecodeProviderSpecAndSecret(machineClass, &v1.Secret{Data: secretData})
+	if err != nil {
+		return err
+	}
+	r.ResourceGroup = providerSpec.ResourceGroup
+
+	clients, err := getAzureClients(secretData)
+	if err != nil {
+		return err
+	}
+
+	instances, err := getVMsWithTag(clients, clusterTag, clusterTagValue, machineClass, r.ResourceGroup, secretData)
+	if err == nil {
+		r.InitialInstances = instances
+		volumes, err := getAvailableDisks(clients, clusterTag, clusterTagValue, r.ResourceGroup)
+		if err == nil {
+			r.InitialVolumes = volumes
+			r.InitialMachines, err = getMachines(machineClass, secretData)
+			return err
+		}
+		return err
+	}
+	return err
+}
+
+// IsOrphanedResourcesAvailable checks whether there are any orphaned resources left.
+// If yes, then prints them and returns true. If not, then returns false
+func (r *ResourcesTrackerImpl) IsOrphanedResourcesAvailable() bool {
+	afterTestExecutionInstances, afterTestExecutionAvailVols, afterTestExecutionAvailmachines, err := r.probeResources()
+	//Check there is no error occured
+	if err == nil {
+		orphanedResourceInstances := differenceOrphanedResources(r.InitialInstances, afterTestExecutionInstances)
+		if orphanedResourceInstances != nil {
+			fmt.Println("orphaned instances are:", orphanedResourceInstances)
```

Comment:
we should change these lines to past tense , like "orphan instances were" etc , because we have already deleted them
This will help the user understand that he doesn't have to go to the provider to delete them.